### ISSUE TITLE
Allow user-controlled private keys

### ISSUE THREAD

**Issue by jas4711** [tillitis/tillitis-key1-apps#94]
While #84 would be nice for me, I realize some use-cases that requires me to control the private key generation myself, and would thus not even want to generate private keys on the device at all.   I couldn't find a feature request to support this, so I'm opening this for discussion.  I think the simplest approach is to resolve #34 and then "import" private keys into the device by encrypting them once, and then later supply the encrypted blob which is decrypted by the device, and its private-key operation is performed, and the output returned.  Thoughts?

**Comment by udf2457** [comment]
Unless I misunderstand you, aren't you basically just suggesting "implement HSM-style CKM AES key wrapping" ?

**Comment by jas4711** [comment]
SSH keys are Ed25519 and not AES keys, but yes, some similar functionality is what I would like.  I want to generate my Ed25519 key using a randomness generator that I can audit, rather than being forced to trust the Tillitis RNG.

**Comment by udf2457** [comment]
True, but the AES is just for wrapping.  I have a YubiHSM and you can optionally export keys in an AES wrapped state, the keys can be in any of the algorithms supported.  The YubiHSM just follows the NIST AES-CCM standard.
